drivers/i2c/i2c_mchp_xec.c

```diff
@@ -120,19 +195,24 @@ static int wait_completion(uint32_t ba)
 		ret = xec_spin_yield(&counter);
 
 		if (ret < 0) {
-			recover_from_error(ba);
-			return ret;
+			if (MCHP_I2C_SMB_STS_RO(ba) & MCHP_I2C_SMB_STS_PIN) {
```

Comment:
PIN=0 is already checked when entering the loop so why is been re-checked  
Is this check for debug only or to check one last time if PIN is asserted before attempting recovery?

Reply:
Yes. One last check before attempting recovery. This is because i saw it was 0 when i printed it within the recovery function. Not sure it helped as i continue to see it as 0 while inside the recovery function sometimes. Guess it cannot be eliminated totally.